core/templates/domain/summary/exploration-summary-backend-api.service.ts

```diff
@@ -17,77 +17,91 @@
  * from the backend.
  */
 
-require('services/alerts.service.ts');
-require('services/validators.service.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
 
-angular.module('oppia').factory('ExplorationSummaryBackendApiService', [
-  '$http', '$q', 'AlertsService',
-  'ValidatorsService', 'EXPLORATION_SUMMARY_DATA_URL_TEMPLATE',
-  function(
-      $http, $q, AlertsService,
-      ValidatorsService, EXPLORATION_SUMMARY_DATA_URL_TEMPLATE) {
-    var _fetchExpSummaries = function(
-        explorationIds, includePrivateExplorations, successCallback,
-        errorCallback) {
-      if (!explorationIds.every(ValidatorsService.isValidExplorationId)) {
-        AlertsService.addWarning('Please enter a valid exploration ID.');
+import { AlertsService } from 'services/alerts.service';
+import { ValidatorsService } from 'services/validators.service';
+import { AppConstants } from 'app.constants';
 
-        var returnValue = [];
-        for (var i = 0; i < explorationIds.length; i++) {
-          returnValue.push(null);
-        }
+@Injectable({
+  providedIn: 'root'
+})
+export class ExplorationSummaryBackendApiService {
+  constructor(
+    private httpClient: HttpClient,
+    private alertsService: AlertsService,
+    private validatorsService: ValidatorsService) {}
 
-        if (errorCallback) {
-          errorCallback(returnValue);
-        }
-        return;
+  _fetchExpSummaries(
+      explorationIds: string[],
+      includePrivateExplorations: boolean,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: string[]) => void): void {
+    if (!explorationIds.every(expId =>
+      this.validatorsService.isValidExplorationId(expId, true))) {
+      this.alertsService.addWarning('Please enter a valid exploration ID.');
+      var returnValue = [];
```

Comment:
let/const instead of var, here and everywhere else in this file.

Reply:
done!